File: java/src/main/java/org/hackerrank/CountApplesAndOranges.java
package org.hackerrank;

import java.util.List;

public class CountApplesAndOranges {
    public static void countApplesAndOranges(int s, int t, int a, int b, List<Integer> apples, List<Integer> oranges) {
        int appleCount = 0;
        int orangeCount = 0;
        for (int apple :
                apples) {
            int applePosition = a + apple;
            if (applePosition >= s && applePosition <= t) {
                appleCount++;
            }
        }
        for (int orange :
                oranges) {
            int orangePosition = b + orange;
            if (orangePosition >= s && orangePosition <= t) {
                orangeCount++;
            }
        }
        System.out.println(appleCount);
        System.out.println(orangeCount);
    }
}

File: js/src/countApplesAndOranges.js
function countApplesAndOranges(s, t, a, b, apples, oranges) {
	// Write your code here
	let appleCount = 0;
	let orangeCount = 0;
	apples.forEach((element) => {
		const applePosition = a + element;
		if (applePosition >= s && applePosition <= t) {
			appleCount++;
		}
	});
	oranges.forEach((element) => {
		const orangePosition = b + element;
		if (orangePosition >= s && orangePosition <= t) {
			orangeCount++;
		}
	});
    return [appleCount, orangeCount];
}

module.exports = countApplesAndOranges;